Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/QuickPulseProcessProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib;
+
+    /// <summary>
+    /// Top CPU collector.
+    /// </summary>
+    internal sealed class QuickPulseProcessProvider : IQuickPulseProcessProvider
+    {
+        private readonly IQuickPulsePerfLib perfLib;
+
+        public QuickPulseProcessProvider(IQuickPulsePerfLib perfLib)
+        {
+            this.perfLib = perfLib ?? PerfLib.PerfLib.GetPerfLib();
```

Comment:
Because that would be an extra constructor. The desired behavior is to use the default IQuickPulsePerfLib implementation whenever nothing is passed. The interface is primarily for unit testing.